packages/theme-check-common/src/checks/schema-presets-block-order/index.ts

```diff
@@ -0,0 +1,150 @@
+import { LiteralNode, ArrayNode } from 'json-to-ast';
+import { getLocEnd, getLocStart, nodeAtPath } from '../../json';
+import { basename } from '../../path';
+import { isBlock, isSection } from '../../to-schema';
+import { JSONNode, LiquidCheckDefinition, Severity, SourceCodeType } from '../../types';
+import { Section } from '../../types/schemas/section';
+import { ThemeBlock } from '../../types/schemas/theme-block';
+import { Context } from '../../types';
+
+export const SchemaPresetsBlockOrder: LiquidCheckDefinition = {
+  meta: {
+    code: 'SchemaPresetsBlockOrder',
+    name: 'Gives recommendations and validations for block_order',
+    docs: {
+      description:
+        'Gives recommendations and validations for block_order for preset blocks as a hash.',
+      recommended: true,
+      url: 'https://shopify.dev/docs/storefronts/themes/tools/theme-check/checks/valid-schema-name',
+    },
+    type: SourceCodeType.LiquidHtml,
+    severity: Severity.WARNING,
+    schema: {},
+    targets: [],
+  },
+
+  create(context) {
+    function getSchema() {
+      const name = basename(context.file.uri, '.liquid');
+      switch (true) {
+        case isBlock(context.file.uri):
+          return context.getBlockSchema?.(name);
+        case isSection(context.file.uri):
+          return context.getSectionSchema?.(name);
+        default:
+          return undefined;
+      }
+    }
+
+    return {
+      async LiquidRawTag(node) {
+        if (node.name !== 'schema' || node.body.kind !== 'json') {
+          return;
+        }
+
+        const offset = node.blockStartPosition.end;
+        const schema = await getSchema();
+        const { validSchema, ast } = schema ?? {};
+        if (!validSchema || validSchema instanceof Error) return;
+        if (!ast || ast instanceof Error) return;
+
+        const presets = validSchema.presets;
+        if (!presets) return;
+
+        presets.forEach((preset, index) => {
+          if ('blocks' in preset && preset.blocks) {
+            checkBlockOrder(preset, context, offset, ast, ['presets', String(index)]);
+          }
+        });
+      },
+    };
+  },
+};
+
+function checkBlockOrder(
+  node: Section.Preset | Section.BlockPresetHash | ThemeBlock.Preset | ThemeBlock.PresetBlockHash,
+  context: Context<SourceCodeType.LiquidHtml>,
+  offset: number,
+  ast: JSONNode,
+  ast_path: string[],
+) {
+  if ('blocks' in node && typeof node.blocks == 'object' && node.blocks) {
+    const blockOrder = 'block_order' in node ? node.block_order : undefined;
+    // check 1: is block_order missing?
+    if (!blockOrder) {
+      if (shouldHaveBlockOrder(node)) {
```

Comment:
Super well done! Only nit is would it be easier on the eyes to place the warning for missing `block_order` on merely the hash key  as opposed to everywhere?
![image](https://github.com/user-attachments/assets/03d34e82-2e4b-4c09-87b2-ff69fb9028b7)